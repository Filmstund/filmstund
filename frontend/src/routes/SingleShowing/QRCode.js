import React, { Component } from "react";
import styled from "styled-components";

import qrcodegen from "../../lib/qrcode";
const QRC = qrcodegen.QrCode;

class QRCode extends Component {
  render() {
    const { value, width, height } = this.props;
    const encodedValue = QRC.encodeText(value, QRC.Ecc.MEDIUM);
    const SwishDiv = styled.div`
      width: ${width};
      height: ${height};
    `;

    return (
      <SwishDiv dangerouslySetInnerHTML={{ __html: encodedValue.toSvgString(5) }} />
    );
  }
}

export default QRCode;
