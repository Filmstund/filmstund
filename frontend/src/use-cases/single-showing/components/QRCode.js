import React, { useMemo } from "react";

import QRC from "../../../lib/qrcode";

const QRCode = ({ value, width, height }) => {
  const svgString = useMemo(
    () => QRC.encodeText(value, QRC.Ecc.MEDIUM).toSvgString(5),
    [value]
  );

  return (
    <div
      style={{ width, height }}
      dangerouslySetInnerHTML={{ __html: svgString }}
    />
  );
};

export default QRCode;
