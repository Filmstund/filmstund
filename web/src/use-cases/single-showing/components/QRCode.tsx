import React, { useMemo } from "react";

import QRC from "../../../lib/qrcode";

interface Props {
  value: string;
  width: string;
  height: string;
}

export const QRCode: React.VFC<Props> = ({ value, width, height }) => {
  const svgString = useMemo(() => {
    // @ts-ignore
    return QRC.encodeText(value, QRC.Ecc.MEDIUM).toSvgString(5);
  }, [value]);

  return (
    <div
      style={{ width, height }}
      dangerouslySetInnerHTML={{ __html: svgString }}
    />
  );
};
