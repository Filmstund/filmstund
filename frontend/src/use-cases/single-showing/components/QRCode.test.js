import React from "react";
import { render } from "react-testing-library";
import QRCode from "./QRCode";

import QRC from "../../../lib/qrcode";

describe("QRCode", () => {
  describe("when provided with a value", () => {
    it("should match the snapshot", () => {
      const { container } = render(
        <QRCode width={100} height={100} value="this is a value" />
      );

      expect(container).toMatchSnapshot();
    });

    it("should memoize the value between renders", () => {
      const encodeTextMock = jest.spyOn(QRC, "encodeText");
      const { rerender } = render(
        <QRCode width={100} height={100} value="this is a value" />
      );

      expect(encodeTextMock).toHaveBeenCalled();

      rerender(<QRCode width={100} height={100} value="this is a value" />);

      expect(encodeTextMock).toHaveBeenCalledTimes(1);

      rerender(
        <QRCode width={100} height={100} value="this is another value" />
      );

      expect(encodeTextMock).toHaveBeenCalledTimes(2);
    });
  });
});
