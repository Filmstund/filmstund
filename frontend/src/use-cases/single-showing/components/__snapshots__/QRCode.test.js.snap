// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCode when provided with a value should match the snapshot 1`] = `
<div>
  <div
    style="width: 100px; height: 100px;"
  >
    <!--?xml version="1.0" encoding="UTF-8"?-->
    


    <svg
      version="1.1"
      viewBox="0 0 35 35"
      xmlns="http://www.w3.org/2000/svg"
    >
      
	
      <rect
        fill="#FFFFFF"
        height="100%"
        stroke-width="0"
        width="100%"
      />
      
	
      <path
        d="M5,5h1v1h-1z M6,5h1v1h-1z M7,5h1v1h-1z M8,5h1v1h-1z M9,5h1v1h-1z M10,5h1v1h-1z M11,5h1v1h-1z M15,5h1v1h-1z M18,5h1v1h-1z M19,5h1v1h-1z M20,5h1v1h-1z M23,5h1v1h-1z M24,5h1v1h-1z M25,5h1v1h-1z M26,5h1v1h-1z M27,5h1v1h-1z M28,5h1v1h-1z M29,5h1v1h-1z M5,6h1v1h-1z M11,6h1v1h-1z M13,6h1v1h-1z M16,6h1v1h-1z M17,6h1v1h-1z M18,6h1v1h-1z M21,6h1v1h-1z M23,6h1v1h-1z M29,6h1v1h-1z M5,7h1v1h-1z M7,7h1v1h-1z M8,7h1v1h-1z M9,7h1v1h-1z M11,7h1v1h-1z M13,7h1v1h-1z M16,7h1v1h-1z M21,7h1v1h-1z M23,7h1v1h-1z M25,7h1v1h-1z M26,7h1v1h-1z M27,7h1v1h-1z M29,7h1v1h-1z M5,8h1v1h-1z M7,8h1v1h-1z M8,8h1v1h-1z M9,8h1v1h-1z M11,8h1v1h-1z M15,8h1v1h-1z M16,8h1v1h-1z M17,8h1v1h-1z M19,8h1v1h-1z M20,8h1v1h-1z M23,8h1v1h-1z M25,8h1v1h-1z M26,8h1v1h-1z M27,8h1v1h-1z M29,8h1v1h-1z M5,9h1v1h-1z M7,9h1v1h-1z M8,9h1v1h-1z M9,9h1v1h-1z M11,9h1v1h-1z M14,9h1v1h-1z M15,9h1v1h-1z M19,9h1v1h-1z M21,9h1v1h-1z M23,9h1v1h-1z M25,9h1v1h-1z M26,9h1v1h-1z M27,9h1v1h-1z M29,9h1v1h-1z M5,10h1v1h-1z M11,10h1v1h-1z M14,10h1v1h-1z M15,10h1v1h-1z M18,10h1v1h-1z M20,10h1v1h-1z M21,10h1v1h-1z M23,10h1v1h-1z M29,10h1v1h-1z M5,11h1v1h-1z M6,11h1v1h-1z M7,11h1v1h-1z M8,11h1v1h-1z M9,11h1v1h-1z M10,11h1v1h-1z M11,11h1v1h-1z M13,11h1v1h-1z M15,11h1v1h-1z M17,11h1v1h-1z M19,11h1v1h-1z M21,11h1v1h-1z M23,11h1v1h-1z M24,11h1v1h-1z M25,11h1v1h-1z M26,11h1v1h-1z M27,11h1v1h-1z M28,11h1v1h-1z M29,11h1v1h-1z M14,12h1v1h-1z M16,12h1v1h-1z M20,12h1v1h-1z M21,12h1v1h-1z M6,13h1v1h-1z M7,13h1v1h-1z M8,13h1v1h-1z M10,13h1v1h-1z M11,13h1v1h-1z M14,13h1v1h-1z M15,13h1v1h-1z M16,13h1v1h-1z M17,13h1v1h-1z M18,13h1v1h-1z M20,13h1v1h-1z M21,13h1v1h-1z M27,13h1v1h-1z M28,13h1v1h-1z M12,14h1v1h-1z M13,14h1v1h-1z M16,14h1v1h-1z M18,14h1v1h-1z M19,14h1v1h-1z M20,14h1v1h-1z M26,14h1v1h-1z M27,14h1v1h-1z M5,15h1v1h-1z M7,15h1v1h-1z M8,15h1v1h-1z M9,15h1v1h-1z M10,15h1v1h-1z M11,15h1v1h-1z M12,15h1v1h-1z M15,15h1v1h-1z M16,15h1v1h-1z M17,15h1v1h-1z M18,15h1v1h-1z M21,15h1v1h-1z M22,15h1v1h-1z M24,15h1v1h-1z M25,15h1v1h-1z M26,15h1v1h-1z M5,16h1v1h-1z M6,16h1v1h-1z M7,16h1v1h-1z M8,16h1v1h-1z M14,16h1v1h-1z M24,16h1v1h-1z M26,16h1v1h-1z M27,16h1v1h-1z M6,17h1v1h-1z M7,17h1v1h-1z M8,17h1v1h-1z M10,17h1v1h-1z M11,17h1v1h-1z M13,17h1v1h-1z M14,17h1v1h-1z M16,17h1v1h-1z M17,17h1v1h-1z M19,17h1v1h-1z M20,17h1v1h-1z M22,17h1v1h-1z M23,17h1v1h-1z M25,17h1v1h-1z M27,17h1v1h-1z M6,18h1v1h-1z M7,18h1v1h-1z M12,18h1v1h-1z M16,18h1v1h-1z M17,18h1v1h-1z M19,18h1v1h-1z M20,18h1v1h-1z M21,18h1v1h-1z M23,18h1v1h-1z M24,18h1v1h-1z M25,18h1v1h-1z M28,18h1v1h-1z M29,18h1v1h-1z M6,19h1v1h-1z M7,19h1v1h-1z M8,19h1v1h-1z M11,19h1v1h-1z M12,19h1v1h-1z M14,19h1v1h-1z M18,19h1v1h-1z M22,19h1v1h-1z M24,19h1v1h-1z M26,19h1v1h-1z M28,19h1v1h-1z M5,20h1v1h-1z M7,20h1v1h-1z M15,20h1v1h-1z M20,20h1v1h-1z M21,20h1v1h-1z M26,20h1v1h-1z M29,20h1v1h-1z M10,21h1v1h-1z M11,21h1v1h-1z M12,21h1v1h-1z M15,21h1v1h-1z M18,21h1v1h-1z M21,21h1v1h-1z M22,21h1v1h-1z M23,21h1v1h-1z M24,21h1v1h-1z M25,21h1v1h-1z M27,21h1v1h-1z M13,22h1v1h-1z M14,22h1v1h-1z M15,22h1v1h-1z M17,22h1v1h-1z M19,22h1v1h-1z M20,22h1v1h-1z M21,22h1v1h-1z M25,22h1v1h-1z M27,22h1v1h-1z M28,22h1v1h-1z M29,22h1v1h-1z M5,23h1v1h-1z M6,23h1v1h-1z M7,23h1v1h-1z M8,23h1v1h-1z M9,23h1v1h-1z M10,23h1v1h-1z M11,23h1v1h-1z M17,23h1v1h-1z M21,23h1v1h-1z M23,23h1v1h-1z M25,23h1v1h-1z M26,23h1v1h-1z M28,23h1v1h-1z M29,23h1v1h-1z M5,24h1v1h-1z M11,24h1v1h-1z M13,24h1v1h-1z M14,24h1v1h-1z M17,24h1v1h-1z M20,24h1v1h-1z M21,24h1v1h-1z M25,24h1v1h-1z M26,24h1v1h-1z M29,24h1v1h-1z M5,25h1v1h-1z M7,25h1v1h-1z M8,25h1v1h-1z M9,25h1v1h-1z M11,25h1v1h-1z M14,25h1v1h-1z M16,25h1v1h-1z M18,25h1v1h-1z M19,25h1v1h-1z M21,25h1v1h-1z M22,25h1v1h-1z M23,25h1v1h-1z M24,25h1v1h-1z M25,25h1v1h-1z M28,25h1v1h-1z M29,25h1v1h-1z M5,26h1v1h-1z M7,26h1v1h-1z M8,26h1v1h-1z M9,26h1v1h-1z M11,26h1v1h-1z M13,26h1v1h-1z M16,26h1v1h-1z M17,26h1v1h-1z M18,26h1v1h-1z M19,26h1v1h-1z M20,26h1v1h-1z M22,26h1v1h-1z M23,26h1v1h-1z M25,26h1v1h-1z M26,26h1v1h-1z M27,26h1v1h-1z M29,26h1v1h-1z M5,27h1v1h-1z M7,27h1v1h-1z M8,27h1v1h-1z M9,27h1v1h-1z M11,27h1v1h-1z M13,27h1v1h-1z M15,27h1v1h-1z M16,27h1v1h-1z M20,27h1v1h-1z M21,27h1v1h-1z M22,27h1v1h-1z M23,27h1v1h-1z M25,27h1v1h-1z M26,27h1v1h-1z M28,27h1v1h-1z M5,28h1v1h-1z M11,28h1v1h-1z M13,28h1v1h-1z M15,28h1v1h-1z M25,28h1v1h-1z M26,28h1v1h-1z M27,28h1v1h-1z M5,29h1v1h-1z M6,29h1v1h-1z M7,29h1v1h-1z M8,29h1v1h-1z M9,29h1v1h-1z M10,29h1v1h-1z M11,29h1v1h-1z M17,29h1v1h-1z M19,29h1v1h-1z M22,29h1v1h-1z M23,29h1v1h-1z M27,29h1v1h-1z M28,29h1v1h-1z M29,29h1v1h-1z"
        fill="#000000"
        stroke-width="0"
      />
      

    </svg>
    

  </div>
</div>
`;
